### PR TITLE
Remove sniffing of HTML

### LINKED ISSUE #173: Should we still sniff if Content-Type is text/html?
I recently investigated a web-platform-test failure (html/semantics/embedded-content/the-iframe-element/iframe_sandbox_anchor_download_block_downloads.tentative.html) in WebKit/CFNetwork and found out that it was caused by sniffing when the Content-Type is text/html:
- https://github.com/web-platform-tests/wpt/pull/41037

This test is passing in Chrome and Firefox so I am wondering. Does the specification (https://mimesniff.spec.whatwg.org/#determining-the-computed-mime-type-of-a-resource) really match Chrome and Firefox here?

Should CFNetwork stop sniffing when the Content-Type is text/html?

### PR DESCRIPTION
**Pull request by annevk**
As no user agent today appears to identify a text/html resource starting with <rss as XML, remove those rules from the standard.

At the same time, make it more clear that XML (and now HTML) are never sniffed. This is a non-normative change for clarity.

Tests: https://github.com/web-platform-tests/wpt/pull/47002.

Closes #173.

- [x] At least two implementers are interested (and none opposed):
   * WebKit
   * Presumably Chromium and Gecko as they don't implement this
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/web-platform-tests/wpt/pull/47002
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: N/A
   * Gecko: N/A
   * WebKit: N/A
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: N/A, MDN doesn't cover this topic at all
- [x] The top of this comment includes a [clear commit message](https://github.com/whatwg/meta/blob/main/COMMITTING.md) to use. 

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***
<a href="https://whatpr.org/mimesniff/192.html" title="Last updated on Jul 4, 2024, 12:36 PM UTC (f23deaa)">Preview</a> | <a href="https://whatpr.org/mimesniff/192/3624957...f23deaa.html" title="Last updated on Jul 4, 2024, 12:36 PM UTC (f23deaa)">Diff</a>

### CODE REVIEW

File: mimesniff.bs [hunk 1]
```diff
@@ -1826,17 +1832,6 @@ algorithm</dfn>:
...
-   If the <a>supplied MIME type</a> is an <a>XML MIME type</a>, the
-   <a>computed MIME type</a> is the <a>supplied MIME type</a>.
-
-   Abort these steps.
-
-  <li>
-   If the <a>supplied MIME type</a>'s <a for="MIME type">essence</a> is "<code>text/html</code>",
-   execute the <a>rules for distinguishing if a resource is a feed or HTML</a> and
-   abort these steps.
-
```
[the remark this answers is not in the document]
Comment (annevk) [reply mimesniff.bs:1839]: As stated in the commit message, for clarity. If we never sniff these types it seems better to make that clear upfront.

[the remark this answers is not in the document]
Comment (GPHemsley) [reply mimesniff.bs:1839]: I think it makes more logical sense where it is; it's the first thing after the anomaly handing. If the no-sniff flag is set, for example, the steps will have already aborted.

[the remark this answers is not in the document]
Comment (annevk) [reply mimesniff.bs:1839]: I think for me the confusing part is really the check-for-apache-bug flag as it's not clear what the supplied MIME type can and cannot be. It also seems weird to me to handle unknown MIME types before HTML and XML, but I could live with that I suppose.

[the remark this answers is not in the document]
Comment (annevk) [reply mimesniff.bs:1839]: @GPHemsley does that make sense?

**Review by domenic — APPROVED** [review]

**Pull request merged**